collections_tech_preview notes/example module_utils python imports (#58896)
ansible/ansible: docs/docsite/rst/collections_tech_preview.rst

plugins directory
------------------

> Add a 'per plugin type' specific subdirectory here, including ``module_utils`` which is usable not only by modules, but by any other plugin by using their Fully Qualified Collection Name (FQCN). This is a way to distribute modules, lookups, filters, and so on, without having to import a role in every play.

Reviewer: FQCN was already defined in an earlier section of this doc so doesn't need to be repeated at all. Just us FQCN, unless you are seeing people not understanding this term and we shouldn't use an acronym at all?

Reviewer 2: When I first read it, I assumed it was a typo of FQDN.

module_utils
~~~~~~~~~~~~

> When working with ``module_utils`` in a Collection, the python ``import`` statement will need to take into account the Fully Qualified Collection Name (FQDN) along with the ``ansible_collections`` convention. The resulting import will look like ``from ansible_collections.{namespace}.{collection}.plugins.module_utils.{util} import {something}``

Reviewer 3: ```suggestion
When working with ``module_utils`` in a Collection, the python ``import`` statement will need to take into account the Fully Qualified Collection Name (FQCN) along with the ``ansible_collections`` convention. The resulting import will look like ``from ansible_collections.{namespace}.{collection}.plugins.module_utils.{util} import {something}``
```

> When working with ``module_utils`` in a Collection, the python ``import`` statement will need to take into account the Fully Qualified Collection Name (FQCN) along with the ``ansible_collections`` convention. The resulting import will look like ``from ansible_collections.{namespace}.{collection}.plugins.module_utils.{util} import {something}``

Reviewer: ```suggestion
When working with ``module_utils`` in a collection, the Python ``import`` statement will need to take into account the FQCN along with the ``ansible_collections`` convention. The resulting import will look like ``from ansible_collections.{namespace}.{collection}.plugins.module_utils.{util} import {something}``
```

> Example snippet of a module using both default Ansible ``module_utils`` and

Reviewer: ```suggestion
The following example snippet shows a module using both default Ansible ``module_utils`` and
```

> those provided by a collection. In this example the Collection is

Reviewer: ```suggestion
those provided by a collection. In this example the collection is
```

``ansible_security``, the namespace is ``community``, and the ``module_util`` in

> question is called ``qradar`` such that the Fully Qualified Collection Name

Reviewer 3: You've spelled out Fully Qualified Collection Name and introduced the acronym more than once . . . you can use the acronym alone now, certainly for the rest of the section, probably for the rest of the page.

Reviewer: I'm reluctant to have `ansible_security` as a community collection example since we may actually have an ansible_security collection from RH. Can we change it to example_security, or something less likely to ever exist?

Author: @samccann we can even just have `example_collection` or whatever ... that was just something I happened to have open in vim at the time so I cribbed some code.
